### DISCUSSION TITLE
Change pitch of the note currently played

### DISCUSSION THREAD

**Discussion by avf** [djipco/webmidi#393] [Q&A] [answered]
Thanks for creating this awesome library!

I have a digital piano connected and I'd like to dynamically change the pitches of the notes being played. I'd like the sound of the changed notes to come from the piano speakers. So for example, transpose all notes being played on the piano by an octave up.

I've attempted to do this by adding a listener, and forwarding the note to the output on the piano, but of course that just results in _both_ notes being played at the same time:

```
const output = WebMidi.outputs[0]; // same device
const input = WebMidi.inputs[0]; // same device
input.addListener('noteon', (e) => {
    output.sendAllSoundOff();
    const pitchChange = 4;
    output.channels[1].playNote(e.note.number + pitchChange, {duration: 100});
});
```
I'm assuming that `output.sendAllSoundOff();` doesn't actually do anything here, because the sound of the original, unaltered note is actually coming from the _input_, not the output.

Is there a way to suppress the original sound from the piano, and _only_ play the sound on the output? Or is there another way to achieve what I'm trying to do?

**Comment by takyano** [comment, marked as answer]
Does your digital piano have an option called LOCAL OFF?

Brand and model of the digital piano would be helpful.

**Reply by avf** [reply]
Thanks for the quick response!

Unfortunately, my Yamaha P-45 doesn't allow Local Control to be turned off, according to [the manual (p18)](https://www.mietklaviere.at/wp-content/uploads/2018/11/p45_de_om_a0.pdf).

However, I found [this forum post](https://www.extreamsd.com/forum/thread-379-post-1042.html) that says you can still turn it off by sending some specific control change events. Is this something I can try easily with WebMidi or do I need to drop down to the native API?

**Reply by avf** [reply]
Ah, I just realized, there's a `sendLocalControl` message API.

This works, I'm quite happy :)

```
const output = WebMidi.outputs[0]; // same device
const input = WebMidi.inputs[0]; // same device
output.sendLocalControl(false);
input.addListener('noteon', (e) => {
    const pitchChange = 4;
    output.channels[1].playNote(e.note.number + pitchChange, {duration: 100});
});
```

Still needs a few tweaks, such as using `sendNoteOn` instead of `playNote`, and forwarding the correct `attack` and `release`, but it's a working proof of concept.
